src/mono/mono/component/marshal-ilgen.h

```diff
@@ -19,21 +36,20 @@ typedef struct {
 	int (*emit_marshal_custom) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
 	int (*emit_marshal_asany) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
 	int (*emit_marshal_handleref) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
-} MonoMarshalIlgenCallbacks;
+} MonoMarshalILgenCallbacks;
+
+MONO_COMPONENT_EXPORT_ENTRYPOINT
+MonoComponentMarshalILgen* mono_component_marshal_ilgen_init (void);
 
 void
-mono_install_marshal_callbacks_ilgen (MonoMarshalIlgenCallbacks *cb);
+mono_install_marshal_callbacks_ilgen (MonoMarshalILgenCallbacks *cb);
```

Comment:
This function is a detail of the `marshal-ilgen` component, and not expected to be used externally.  I would make a new `marshal-ilgen-internals.h` header in this directory and put the declaration in there.